prime-router/src/main/kotlin/history/azure/DatabaseDeliveryAccess.kt

```diff
@@ -141,4 +155,88 @@ class DatabaseDeliveryAccess(
             query.fetchInto(DeliveryFacility::class.java)
         }
     }
+
+    /**
+     * Get a list of matching delivery facilities for reports delivered to a given [receiver]
+     * @todo further work on https://github.com/CDCgov/prime-reportstream/issues/9272
+     * @param receiver the organization that is being checked.
+     * @param receivingOrgSvc is a specifier for the receiving organization's service.
+     * @param sortDir sort the table by date in ASC or DESC order
+     * @param sortColumns sort the table by a specific set of columns
+     * @param since is the OffsetDateTime minimum date to get results for.
+     * @param until is the OffsetDateTime maximum date to get results for.
+     * @return List of matching delivery facilities
+     */
+    fun fetchBulkDeliveryFacilities(
+        receiver: String,
+        receivingOrgSvc: String?,
+        sortDir: SortDir,
+        sortColumns: List<BulkFacilitySortColumn>,
+        since: OffsetDateTime?,
+        until: OffsetDateTime?,
+    ): List<DeliveryFacility> {
+        val reportIds = db.transactReturning { txn ->
+            var filter = this.organizationFilter(receiver, receivingOrgSvc)
+            if (since != null) {
+                filter = filter.and(ACTION.CREATED_AT.ge(since))
+            }
+            if (until != null) {
+                filter = filter.and(ACTION.CREATED_AT.lt(until))
+            }
+
+            val query = DSL.using(txn)
+                // Note the report file and action tables have columns with the same name, so we must specify what we need.
+                .select(REPORT_FILE.REPORT_ID)
+                .from(
+                    ACTION.join(REPORT_FILE).on(
+                        REPORT_FILE.ACTION_ID.eq(ACTION.ACTION_ID)
+                    )
+                )
+                .where(filter)
+                .groupBy(REPORT_FILE.REPORT_ID)
+
+            query.fetchInto(ReportId::class.java)
+        }
+
+        return db.transactReturning { txn ->
+            val sortedColumns = sortColumns.map {
+                val column = when (it) {
+                    /* Decides sort column by enum */
+                    BulkFacilitySortColumn.DATE -> ITEM_LINEAGE.CREATED_AT
+                    BulkFacilitySortColumn.PROVIDER -> COVID_RESULT_METADATA.ORDERING_PROVIDER_NAME
+                    BulkFacilitySortColumn.FACILITY -> COVID_RESULT_METADATA.TESTING_LAB_NAME
+                    BulkFacilitySortColumn.SUBMITTER -> COVID_RESULT_METADATA.SENDER_ID
+                    BulkFacilitySortColumn.REPORT_ID -> COVID_RESULT_METADATA.REPORT_ID
+                }
+
+                when (sortDir) {
+                    /* Applies sort order by enum */
+                    SortDir.ASC -> column.asc()
+                    SortDir.DESC -> column.desc()
+                }
+            }
+
+            val childReportIds: Field<String> = DSL.field(
+                "item_lineage.child_report_id",
+                String::class.java
```

Comment:
Why cast to a string here?

Reply:
I was using an overly long declaration. It's simplified now